### PR TITLE
Add types for eth-url-parser

### PR DESCRIPTION
**Pull request by gomesalexandre**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## eth-url-parser ([<kbd>unpkg</kbd>](https://unpkg.com/browse/eth-url-parser@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'eth-url-parser'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.
1. apply
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4189e924e828dfe8804fa9d17f2a5ea5d077e6d7
</p>

**Comment by gomesalexandre** [comment]
Opening for early review, the types are working but the tests are somehow not passing, hinting at `export =` syntax, but the declaration-level exports should be matching the library, according to both DefinitelyTyped and TypeScript docs.
See thread on Discord:

https://discord.com/channels/508357248330760243/1023217663608094720

File: types/eth-url-parser/tslint.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    "npm-naming": [
+      true,
+      {
+        "mode": "code",
+        "errors": [
+          ["NeedsExportEquals", false]
+        ]
+      }
+    ]
+  }
```
Comment (peterblazejewicz) [inline types/eth-url-parser/tslint.json:13]:
```suggestion
  "extends": "@definitelytyped/dtslint/dt.json"
```
why do you need those exclusions? Your code looks OK and should work without those

Comment (gomesalexandre) [reply types/eth-url-parser/tslint.json:13]: Thanks for the review @peterblazejewicz! This is a current issue with dtslint's `NeedExportEquals` rule, see https://github.com/DefinitelyTyped/DefinitelyTyped/issues/62444 and the matching Discord thread https://discord.com/channels/508357248330760243/757992230077333655/1023217663608094720.

Confirmed by @sandersn on Discord the types are sane and this is a false positive. Just gave it another try reverting the rule ignore and unfortunately, the issue is still happening on latest master so we still need to disable the rule.

**Review by jakebailey — APPROVED** [review]

**Comment by gomesalexandre** [comment]
Ready to merge

**Pull request merged**